### PR TITLE
Add ceilometer config customization

### PR DESCRIPTION
**Pull request by vyzigold**
This lets us customize the ceilometer configuration based on the customServiceConfig field in ceilometer's part of the control plane CR.

Depends-On: https://github.com/openstack-k8s-operators/telemetry-operator/pull/285

### CODE REVIEW

**Comment by vyzigold** [comment]
recheck FAILED! => {"msg": "'ctlplane_cidr' is undefined. 'ctlplane_cidr' is undefined"}

Should apparently be fixed in https://github.com/openstack-k8s-operators/openstack-operator/pull/639

**Comment by vyzigold** [comment]
I'll need to do a few more changes to the ansible to account for a possibility of the custom.conf missing.

/hold

**Comment by vyzigold** [comment]
recheck

File: roles/edpm_telemetry/tasks/deploy_ceilometer.yml [hunk 1]
```diff
@@ -30,25 +30,44 @@
...
+      - "src": "{{ edpm_telemetry_config_src }}/polling.yaml"
+        "dest": "{{ edpm_ceilometer_config_dest }}/ceilometer/polling.yaml"
+      - "src": "{{ edpm_telemetry_config_src }}/ceilometer_agent_compute.json"
+        "dest": "{{ edpm_ceilometer_config_dest }}/containers/ceilometer_agent_compute.json"
+      - "src": "{{ edpm_telemetry_config_src }}/ceilometer-agent-compute.json"
+        "dest": "{{ edpm_ceilometer_config_dest }}/ceilometer/ceilometer-agent-compute.json"
+      - "src": "{{ edpm_telemetry_config_src }}/ceilometer_agent_ipmi.json"
+        "dest": "{{ edpm_ceilometer_config_dest }}/containers/ceilometer_agent_ipmi.json"
+      - "src": "{{ edpm_telemetry_config_src }}/ceilometer-agent-ipmi.json"
+        "dest": "{{ edpm_ceilometer_config_dest }}/ceilometer/ceilometer-agent-ipmi.json"
```
Comment (jlarriba) [inline roles/edpm_telemetry/tasks/deploy_ceilometer.yml:47]: it lacks the edpm_telemetry tag

File: roles/edpm_telemetry/tasks/deploy_ceilometer.yml [hunk 2]
```diff
@@ -30,25 +30,44 @@
...
+  tags:
+    - edpm_telemetry
+
+- name: Append custom.conf to config files
+  ansible.builtin.set_fact:
+    configs: "{{ configs + [{
+      'src': edpm_telemetry_config_src + '/custom.conf',
+      'dest': edpm_ceilometer_config_dest + '/ceilometer/custom.conf' }] }}"
+  when: custom_ceilometer_conf.stat.exists
+
```
Comment (jlarriba) [inline roles/edpm_telemetry/tasks/deploy_ceilometer.yml:63]: same as above

**Review by jlarriba — CHANGES_REQUESTED** [review]

**Review by jlarriba — APPROVED** [review]

**Review by jpodivin — APPROVED** [review]

**Pull request merged**